cmd/whitesourceExecuteScan.go

```diff
@@ -408,25 +439,35 @@ func checkSecurityViolations(config *ScanOptions, scan *ws.Scan, sys whitesource
 	}
 	if config.ProjectToken != "" {
 		project := ws.Project{Name: config.ProjectName, Token: config.ProjectToken}
-		if err := checkProjectSecurityViolations(cvssSeverityLimit, project, sys); err != nil {
+		if _, err := checkProjectSecurityViolations(cvssSeverityLimit, project, sys); err != nil {
 			return err
 		}
 	} else {
+		vulnerabilitiesCount := 0
+		var errorsOccured []string
 		for _, project := range scan.ScannedProjects() {
-			if err := checkProjectSecurityViolations(cvssSeverityLimit, project, sys); err != nil {
-				return err
+			// collect errors and aggregate vulnerabilities from all projects
+			if vulCount, err := checkProjectSecurityViolations(cvssSeverityLimit, project, sys); err != nil {
+				vulnerabilitiesCount += vulCount
+				errorsOccured = append(errorsOccured, fmt.Sprint(err))
+			}
+		}
+		if len(errorsOccured) > 0 {
+			if vulnerabilitiesCount > 0 {
+				log.SetErrorCategory(log.ErrorCompliance)
 			}
+			return fmt.Errorf(strings.Join(errorsOccured, ": "))
```

Comment:
All this should also happen for line 442. There is currently a duplication in how projects are scanned. When using an aggregate scan (always the case when using the UA?), then the performed scans will not be reflected in `scan.ScannedProjects`. Instead, `config.ProjectToken` must be used to fetch the scan results. Eventually, this should be fixed by also using `scan.ScannedProjects` when performing an aggregate scan, so that the subsequent code-paths do not need to be aware of this unintuitive debt.

Reply:
I've left another comment regarding a TODO where the UA codepath uses `AppendScannedProject()`. It should be checked if there is at all a code-path left where AppendScannedProject() is not used. If all scans are reflected in `scan.ScannedProjects`, then this `if` can be removed (only keep `else` path).

Reply:
as far as I understand the first codepath is only used for aggregation into one project which is to be avoided anyway according to our experts.
That is why it has been left out for now.
I agree that we should remove this implicit handling at some point and make it rather explicit.